llvmcpy/__init__.py

```diff
@@ -0,0 +1,77 @@
+import hashlib
+import importlib.util
+import os
+import subprocess
+import sys
+from pathlib import Path
+from shutil import which
+from typing import List, Optional
+
+import platformdirs
+
+from ._generator import Generator
+
+
+class LLVMCpy:
+    def __init__(self, llvm_config: Optional[str] = None):
+        self._search_paths = os.environ.get("PATH", os.defpath).split(os.pathsep)
+        if llvm_config is not None:
+            self._llvm_config = llvm_config
+        else:
+            self._llvm_config = self._find_program("LLVM_CONFIG", ["llvm-config"])
+        self._search_paths.insert(0, self._run_llvm_config(["--bindir"]))
+        self.version = self._run_llvm_config(["--version"])
+
+        module = self._get_module()
+        for elem in dir(module):
+            setattr(self, elem, getattr(module, elem))
```

Comment:
I think I suggested not to use `setattr` and have instead `__new__` return transparently an instance from `_generated.py` of `LLVMCPy`.
What was of that idea?

Reply:
1. The `if True:` was dropped, but adding `class LLVMCPyImpl` requires either adding `self` to all the instances or adding `@staticmethod`, both of which are not trivial. Also, if going for that route, it's not enough to add the global functions, also the classes need to be nested inside. When inside the class it's not possible to reference them plainly, so every use of a class would need to be prefixed with `self.__class__.<class>`
2. Sure, but then it would inherit the `__new__` so you need to patch the `__new__` in the child class to invoke `type` directly

Reply:
OK, let's postpone all of this to some time when we'll try and see if we can actually support mypy/autocomplete.